remotes/docker/fetcher.go

```diff
@@ -52,6 +52,7 @@ func (r dockerFetcher) Fetch(ctx context.Context, desc ocispec.Descriptor) (io.R
 
 		if resp.StatusCode > 299 {
 			if resp.StatusCode == http.StatusNotFound {
+				resp.Body.Close()
 				continue // try one of the other urls.
 			}
 			resp.Body.Close()
```

Comment:
You could also just move this existing one up above the conditional.

Reply:
Update this. Thanks a lot for your feedback.